src/Microsoft.ML.TorchSharp/NasBert/NerTrainer.cs

```diff
@@ -0,0 +1,363 @@
+
+[assembly: LoadableClass(typeof(NerTransformer), null, typeof(SignatureLoadModel),
+    NerTransformer.UserName, NerTransformer.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(NerTransformer), null, typeof(SignatureLoadRowMapper),
+    NerTransformer.UserName, NerTransformer.LoaderSignature)]
+
+namespace Microsoft.ML.TorchSharp.NasBert
+{
+    using TargetType = VBuffer<long>;
+
+    /// <summary>
+    /// The <see cref="IEstimator{TTransformer}"/> for training a Deep Neural Network(DNN) to classify text.
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    /// To create this trainer, use [NER](xref:Microsoft.ML.TorchSharpCatalog.NameEntityRecognition(Microsoft.ML.MulticlassClassificationCatalog.MulticlassClassificationTrainers,System.String,System.String,System.String,Int32,Int32,Int32,Microsoft.ML.TorchSharp.NasBert.BertArchitecture,Microsoft.ML.IDataView)).
+    ///
+    /// ### Input and Output Columns
+    /// The input label column data must be a Vector of [string](xref:Microsoft.ML.Data.TextDataViewType) type and the sentence columns must be of type<xref:Microsoft.ML.Data.TextDataViewType>.
+    ///
+    /// This trainer outputs the following columns:
+    ///
+    /// | Output Column Name | Column Type | Description|
+    /// | -- | -- | -- |
+    /// | `PredictedLabel` | Vector of [key](xref:Microsoft.ML.Data.KeyDataViewType) type | The predicted label's index. If its value is i, the actual label would be the i-th category in the key-valued input label type. |
+    /// |  |  |
+    /// | -- | -- |
+    /// | Machine learning task | Multiclass classification |
+    /// | Is normalization required? | No |
+    /// | Is caching required? | No |
+    /// | Required NuGet in addition to Microsoft.ML | Microsoft.ML.TorchSharp and libtorch-cpu or libtorch-cuda-11.3 or any of the OS specific variants. |
+    /// | Exportable to ONNX | No |
+    ///
+    /// ### Training Algorithm Details
+    /// Trains a Deep Neural Network(DNN) by leveraging an existing pre-trained NAS-BERT roBERTa model for the purpose of name entity recognition.
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    ///
+    public class NerTrainer : NasBertTrainer<VBuffer<uint>, TargetType>
+    {
+        internal NerTrainer(IHostEnvironment env, NasBertOptions options) : base(env, options)
+        {
+        }
+
+        internal NerTrainer(IHostEnvironment env,
+            string labelColumnName = DefaultColumnNames.Label,
+            string predictionColumnName = DefaultColumnNames.PredictedLabel,
+            string sentence1ColumnName = "Sentence1",
+            int batchSize = 32,
+            int maxEpochs = 10,
+            IDataView validationSet = null,
+            BertArchitecture architecture = BertArchitecture.Roberta) :
+            this(env, new NasBertOptions
+            {
+                PredictionColumnName = predictionColumnName,
```

Comment:
@LittleLittleCloud @zewditu - any thoughts/questions on NerTrainer?

Does this align with our advance training options for the other deep learning scenarios?

Reply:
I changed the options class from being NasBertOptions to be NerOptions. I'll resolve this comment for now, but please open another one if there are other questions.